File: lab1.js
const readline = require('readline');

function caesarCipherEncrypt(message, key) {
  let encryptedMessage = '';
  for (let i = 0; i < message.length; i++) {
    const char = message[i];
    if (char.match(/[a-z]/i)) {
      const code = message.charCodeAt(i);
      let shift = key % 26;
      if (char.match(/[a-z]/)) {
        if (code + shift > 122) {
          encryptedMessage += String.fromCharCode(code + shift - 26);
        } else {
          encryptedMessage += String.fromCharCode(code + shift);
        }
      } else if (char.match(/[A-Z]/)) {
        if (code + shift > 90) {
          encryptedMessage += String.fromCharCode(code + shift - 26);
        } else {
          encryptedMessage += String.fromCharCode(code + shift);
        }
      }
    } else {
      encryptedMessage += char;
    }
  }
  return encryptedMessage;
}

function caesarCipherDecrypt(encryptedMessage, key) {
  return caesarCipherEncrypt(encryptedMessage, -key);
}

function main() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });

  rl.question("Enter your message: ", (userMessage) => {
    rl.question("Enter the shift key (an integer): ", (data) => {
      data = parseInt(data);

      if (isNaN(data)) {
        console.log("Invalid shift key. Please enter a valid integer.");
        rl.close();
        return;
      }

      const encryptedMessage = caesarCipherEncrypt(userMessage, data);
      const decryptedMessage = caesarCipherDecrypt(encryptedMessage, data);

      console.log(`
        Original Message: ${userMessage}
        Encrypted Message: ${encryptedMessage}
        Decrypted Message: ${decryptedMessage}
      `);

      rl.close();
    });
  });
}

main();

File: practice/C labs/lab1.c
#include <stdio.h>
#include <stdlib.h>
void caesarCipherEncrypt(char *message, int key)
{
    for (int i = 0; message[i] != '\0'; ++i)
    {
        if (message[i] >= 'A' && message[i] <= 'Z')
        {
            message[i] = (message[i] + key - 'A') % 26 + 'A';
        }
        else if (message[i] >= 'a' && message[i] <= 'z')
        {
            message[i] = (message[i] + key - 'a') % 26 + 'a';
        }
    }
}
void caesarCipherDecrypt(char *message, int key)
{
    caesarCipherEncrypt(message, -key);
}
int main()
{
    char message[100];
    int key;
    printf("Enter a message: ");
    fgets(message, sizeof(message), stdin);
    printf("Enter the key: ");
    scanf("%d", &key);
    caesarCipherEncrypt(message, key);
    printf("Encrypted Message: %s", message);
    caesarCipherDecrypt(message, key);
    printf("Decrypted Message: %s", message);
    return 0;
}
